ssl/record/rec_layer_s3.c

```diff
@@ -533,7 +533,7 @@ int ssl_release_record(SSL_CONNECTION *s, TLS_RECORD *rr, size_t length)
  * Return up to 'len' payload bytes received in 'type' records.
  * 'type' is one of the following:
  *
- *   -  SSL3_RT_HANDSHAKE (when ssl3_get_message calls us)
+ *   -  SSL3_RT_HANDSHAKE (when tls_get_message_header calls us)
```

Comment:
In this case its `tls_get_message_header` or `tls_get_message_body` (they both make such a call):

https://github.com/openssl/openssl/blob/0f9caad5b95e901b87fe45cf85c9582071ca0b23/ssl/statem/statem_lib.c#L1530-L1546

and

https://github.com/openssl/openssl/blob/0f9caad5b95e901b87fe45cf85c9582071ca0b23/ssl/statem/statem_lib.c#L1640-L1657

Reply:
Yeah even I was confused at this point. But I thought that since tls_get_message_header is calling the ssl_read_bytes. I can go for it. But honestly I do not see either of them calling "ssl3" to be specific. It might be deep down in the nested calls. I tried looking for it but didnt find much.

sumitra@sumitra:~/opensource/openssl$ git grep 'ssl3_read_bytes' -- .
CHANGES.md:   ssl3_read_bytes() found application data while handshake
CHANGES.md: * Fix ssl3_read_bytes (ssl/s3_pkt.c): To ignore messages of unknown
CHANGES.md:   A 'peek' parameter has also been added to ssl3_read_bytes, which
doc/man3/ERR_put_error.pod:descriptions. For example, the function ssl3_read_bytes() reports a
ssl/record/rec_layer_s3.c:int ssl3_read_bytes(SSL *ssl, uint8_t type, uint8_t *recvd_type,
ssl/record/record.h:__owur int ssl3_read_bytes(SSL *s, uint8_t type, uint8_t *recvd_type,
ssl/s3_lib.c:         * ssl3_read_bytes decided to call s->handshake_func, which called
ssl/s3_lib.c:         * ssl3_read_bytes to read handshake data. However, ssl3_read_bytes
ssl/ssl_local.h:                ssl3_read_bytes, \
ssl/ssl_local.h:                ssl3_read_bytes, \


You can see there is no direct call to ssl3_read_bytes anywhere.

PS: I am new to open source. I apologise for the mistakes made.

Reply:
Okay. I got it thanks.

I will then mention both the functions tls_get_message_header and tls_get_message_body in the comments.